sql/src/main/java/io/crate/expression/scalar/StringToArrayFunction.java

```diff
@@ -0,0 +1,162 @@
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+public class StringToArrayFunction extends Scalar<String[], String> {
+
+    private static final String NAME = "string_to_array";
+
+    private final FunctionInfo info;
+
+    private StringToArrayFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(NAME,
+            new BaseFunctionResolver(
+                FuncParams
+                    .builder(Param.STRING, Param.STRING)
+                    .withVarArgs(Param.STRING).limitVarArgOccurrences(1)
+                    .build()) {
+
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> dataTypes) {
+                    return new StringToArrayFunction(
+                        new FunctionInfo(
+                            new FunctionIdent(NAME, dataTypes),
+                            DataTypes.STRING_ARRAY
+                        )
+                    );
+                }
+            });
+    }
+
+    @Override
+    public String[] evaluate(TransactionContext txnCtx, Input<String>[] args) {
+        assert args.length == 2 || args.length == 3 : "number of args must be 2 or 3";
+
+        String str = args[0].value();
+        if (str == null) {
+            return null;
+        }
+        if (str.isEmpty()) {
+            return new String[0];
+        }
+
+        String separator = args[1].value();
+        String nullStr = null;
+        if (args.length == 3) {
+            nullStr = args[2].value();
+        }
+
+        return split(str, separator, nullStr);
+    }
+
+    /**
+     * <p>Splits a {@code String} into an array of strings using a separator
+     * string specified and a null-string to set the substring to {@code null}
+     * if they match.
+     * </p>
+     *
+     * <p>
+     * A {@code null} separator splits the string into array in which each
+     * string characters becomes a separate element. If the separator is an
+     * empty string, then the input string is returned as a one element array.
+     * </p>
+     *
+     * @param str       The input {@code String} to split.
+     * @param separator The separator {@code String} used for splitting
+     *                  the input string into substrings. May be {@code null}.
+     * @param nullStr   The null string used to set substrings to {@code null}
+     *                  if they match. May be {@code null}.
+     * @return An array of {@code String}.
+     */
+    private String[] split(
+        @Nonnull String str,
+        @Nullable String separator,
+        @Nullable String nullStr) {
+
+        ArrayList<String> subStrings = new ArrayList<>(str.length());
```

Comment:
`str.length()` doesn't really fit, does it? I think this could be moved into the according branches. For example in the `separator.isEmpty()` case it is never used.

In the `separator == null` check the initialization would also fit.